File: backend/internal/identity/domain/avatar.go
package domain

type AvatarID string

const (
	Avatar1 AvatarID = "avatar_1"
	Avatar2 AvatarID = "avatar_2"
	Avatar3 AvatarID = "avatar_3"
	Avatar4 AvatarID = "avatar_4"
	Avatar5 AvatarID = "avatar_5"
)

var AvailableAvatarIDs = []AvatarID{Avatar1, Avatar2, Avatar3, Avatar4, Avatar5}

func IsValidAvatarID(id AvatarID) bool {
	for _, avatar := range AvailableAvatarIDs {
		if avatar == id {
			return true
		}
	}
	return false
}

File: ios/MuseApp/Domain/Entities/Avatar.swift
import Foundation

public struct Avatar: Equatable, Sendable {
    public let id: String
    public let displayName: String

    public init(id: String, displayName: String) {
        self.id = id
        self.displayName = displayName
    }
}

public enum AvatarCatalog {
    public static let all: [Avatar] = [
        Avatar(id: "avatar_1", displayName: "Avatar 1"),
        Avatar(id: "avatar_2", displayName: "Avatar 2"),
        Avatar(id: "avatar_3", displayName: "Avatar 3"),
        Avatar(id: "avatar_4", displayName: "Avatar 4"),
        Avatar(id: "avatar_5", displayName: "Avatar 5")
    ]
}
